sdk/web-pubsub/web-pubsub-express/review/web-pubsub-express.api.md

```diff
@@ -50,8 +51,9 @@ export interface ConnectResponse {
 // @public
 export interface ConnectResponseHandler {
     fail(code: 400 | 401 | 500, detail?: string): void;
```

Comment:
how about

`fail(response: ErrorResponse | MqttErrorResponse)`
`interface ErrorResponse { code: 400|401|500, detail?: string }`

Reply:
This will be a breaking change since it modifies the existing `fail` handler. Would it be okay?

Reply:
Would overloads like this work?
```
fail(code: 400 | 401 | 500, detail?: string): void;
fail(response: ConnectErrorResponse | MqttConnectErrorResponse): void;
```

Reply:
already tried this and it's giving this error since these are functions within an object, and their names can not be the same since they are keys within an object
<img width="440" alt="image" src="https://github.com/user-attachments/assets/667beda2-8c3f-4016-a80d-233f3d90d67a">

Reply:
The actual implementation would just have a signature of 
```ts
fail(codeOrResponse: 400 | 401 | 500 | ConnectErrorResponse | MqttConnectErrorResponse, detail?: string)`
```

Reply:
This would be confusing to customer since the `detail` is already included in `ConnectErrorResponse` and also is not relevant if the response is `MqttConnectErrorResponse`, since it uses the `reason` field

Reply:
> The actual implementation would just have a signature of
> 
> ```ts
> fail(codeOrResponse: 400 | 401 | 500 | ConnectErrorResponse | MqttConnectErrorResponse, detail?: string)`
> ```

We also added logic to support MQTT in the original fail() handler. This new failWith() handler is added in case if users want to have more fine-grained control over the MQTT return code